school.c

```diff
@@ -0,0 +1,743 @@
+
+    printf("Enter the first name of the student to edit: ");
+    get_input(firstName, 50);
+    printf("Enter the last name of the student to edit: ");
+    get_input(lastName, 50);
+
+    // Find the student in the school system
+    Student* targetStudent = findStudent(firstName, lastName);
+    if (targetStudent == NULL) {
+        printf("Student not found.\n");
+        return;
+    }
+
+    int subjectChoice;
+    printf("Select the subject to edit the grade:\n");
+    printf("[0] Biology\n");
+    printf("[1] Algebra\n");
+    printf("[2] Chemistry\n");
+    printf("[3] Calculus\n");
+    printf("[4] English\n");
+    printf("[5] History\n");
+    printf("[6] Physics\n");
+    printf("[7] Grammar\n");
+    printf("[8] Geography\n");
+    printf("[9] Statistics\n");
+    printf("Enter the subject number (0-9): ");
+    subjectChoice = get_int_input();
+
+    while (subjectChoice < 0 || subjectChoice > 9) {
+        printf("Invalid subject number.\n");
+        subjectChoice = get_int_input();
+    }
+
+    int newGrade;
+    printf("Enter the new grade for the selected subject: ");
+    newGrade = get_int_input();
+
+    while (newGrade < 0 || newGrade > 100) {
+        printf("Invalid grade. Grade should be between 0 and 100.\n");
+        newGrade = get_int_input();
+    }
+
+    // Update the grade for the selected subject
+    targetStudent->marks[subjectChoice] = newGrade;
+
+    printf("Grade updated successfully.\n");
+}
+
+// Function to get the top N students in a specific subject in a specific grade
+void printTopNStudentsPerCourse() {
+    int grade, subject, topN;
+
+    printf("Enter the grade (1 to 12): ");
+    grade = get_int_input();
+    while (grade < 1 || grade > NUM_GRADES) {
+        printf("Invalid grade.\n");
+        grade = get_int_input();
+    }
+
+    printf("Select the subject:\n");
+    printf("[0] Biology\n");
+    printf("[1] Algebra\n");
+    printf("[2] Chemistry\n");
+    printf("[3] Calculus\n");
+    printf("[4] English\n");
+    printf("[5] History\n");
+    printf("[6] Physics\n");
+    printf("[7] Grammar\n");
+    printf("[8] Geography\n");
+    printf("[9] Statistics\n");
+    printf("Enter the subject number (0-9): ");
+    subject = get_int_input();
+    while (subject < 0 || subject > 9) {
+        printf("Invalid subject number.\n");
+        subject = get_int_input();
+    }
+
+    printf("Enter the number of top students you want to view: ");
+    topN = get_int_input();
+    while (topN <= 0) {
+        printf("Invalid value for topN.\n");
+        topN = get_int_input();
+    }
+
+    Student* topStudents[topN];
+    for (int i = 0; i < topN; ++i) {
+        topStudents[i] = NULL;
+    }
+
+    for (int classNum = 0; classNum < NUM_CLASSES; ++classNum) {
+        Student* currentStudent = school[grade - 1][classNum];
+        while (currentStudent != NULL) {
+            if (topStudents[topN - 1] == NULL || currentStudent->marks[subject] > topStudents[topN - 1]->marks[subject]) {
```

Comment:
I have seen this implementation for almost all of the girls.
I do not understand why this is useful? shifting all students in the array, instead of
just replacing the new student with the lowest average student.